dev_guide/builds.adoc

```diff
@@ -191,6 +191,84 @@ See the xref:../creating_images/s2i.adoc#creating-images-s2i[S2I Requirements] t
 on how to create a builder image supporting incremental builds.
 ====
 
+
+[[extended-builds]]
+
+=== Extended builds
+
+For compiled languages (Go, C, C++, Java, etc.) the dependencies necessary for
+compilation might increase the size of the image or introduce vulnerabilities
+that can be exploited.
+
+To avoid these problems, source-to-image introduces a two image build process
+that allows an application to be built via the normal flow in a builder image
+but then injects the resulting application artifacts into a runtime-only image
+for execution.
+
+To offer flexibility in this process, source-to-image will execute an
+*_assemble-runtime_* script inside the runtime image that allows further
+customization of the resulting runtime image.
+
+More information about this can be found in the official
+https://github.com/openshift/source-to-image/blob/master/docs/runtime_image.md[source-to-image
+extended builds docs].
+
+This feature is available only for the source strategy.
+====
+[source,yaml]
+----
+strategy:
+  type: "Source"
+  sourceStrategy:
+    from:
+      kind: "ImageStreamTag"
+      name: "builder-image:latest"
+    runtimeImage: <1>
+      kind: "ImageStreamTag"
+      name: "runtime-image:latest"
+    runtimeArtifacts: <2>
+      - sourcePath: "/path/to/source"
+        destinationDir: "path/to/destination"
+----
+<1> The runtime image that the artifacts should be copied to.
+This will be the final image that the application will run on. This image
+should contain the minimum application dependencies to run the injected content
+from the builder image.
+
+<2> The runtime artifacts are a mapping of artifacts produced in the builder
+image that should be injected into the runtime image.
+`sourcePath` can be the full path to a file or directory inside the builder
+image.
+`destinationDir` must be a directory inside the runtime image where the
+artifacts will be copied. This directory is relative to the specified
+`WORKDIR` inside that image.
+
+====
+
+[NOTE]
+====
+In the current implementation you cannot have incremental extended builds thus
+the `incremental` option is not valid with `runtimeImage`.
+====
+
+If the runtime image needs authentication to be pulled across OpenShift
+projects or from another private registry, the details can be specified
+xref:managing_images.adoc#using-image-pull-secrets[within the image pull
+secret config].
+
```

Comment:
Add a unnique anchor before the heading:

[[dev-guide-testing-your-application]]